src/Medidata.ZipkinTracer.Core/ZipkinClient.cs

```diff
@@ -1,67 +1,75 @@
 ﻿using System;
 using System.Runtime.CompilerServices;
 using log4net;
-using Medidata.CrossApplicationTracer;
 using Medidata.ZipkinTracer.Core.Collector;
 
 namespace Medidata.ZipkinTracer.Core
 {
     public class ZipkinClient: ITracerClient
     {
-        internal bool isTraceOn;
         internal SpanCollector spanCollector;
         internal SpanTracer spanTracer;
 
-        private ITraceProvider traceProvider;
         private ILog logger;
+        public bool IsTraceOn { get; set; }
+        public ITraceProvider TraceProvider { get; }
+        public IZipkinConfig ZipkinConfig { get; }
 
-        public ZipkinClient(ITraceProvider tracerProvider, ILog logger) : this(tracerProvider, logger, new ZipkinConfig(), new SpanCollectorBuilder()) { }
+        public ZipkinClient(ITraceProvider tracerProvider, ILog logger, IZipkinConfig zipkinConfig)
+            : this(tracerProvider, logger, zipkinConfig, new SpanCollectorBuilder()) { }
 
-        public ZipkinClient(ITraceProvider traceProvider, ILog logger, IZipkinConfig zipkinConfig, ISpanCollectorBuilder spanCollectorBuilder)
+        public ZipkinClient(ITraceProvider traceProvider, ILog logger, IZipkinConfig zipkinConfig,
+            ISpanCollectorBuilder spanCollectorBuilder)
         {
-            this.logger = logger;
+            if (traceProvider == null) throw new ArgumentNullException(nameof(traceProvider));
+            if (logger == null) throw new ArgumentNullException(nameof(logger));
+            if (zipkinConfig == null) throw new ArgumentNullException(nameof(zipkinConfig));
+            if (spanCollectorBuilder == null) throw new ArgumentNullException(nameof(spanCollectorBuilder));
 
-            isTraceOn = logger != null && IsConfigValuesValid(zipkinConfig) && IsTraceProviderValidAndSamplingOn(traceProvider);
+            IsTraceOn = zipkinConfig.Enable && IsTraceProviderSamplingOn(traceProvider);
 
-            if (!isTraceOn)
+            if (!IsTraceOn)
                 return;
 
+            zipkinConfig.Validate();
```

Comment:
Somehow Validate() here is the 2nd call when instantiated from ZipkinMiddleWare. We can call this once before ZipkinClient is instantiated.